### PR TITLE
Refresh Web Agent

### PR DESCRIPTION
**Pull request by zahnno**
PR includes: 

- Web agent decorator which provides a refresh rate to keep up to date values on decorated web agents.
- Memory Web Agent decorated by RefreshWebAgent. 
- Dependant on https://github.com/foam-framework/foam2/pull/845

### CODE REVIEW

**Comment by marcogbarcellos** [comment]
lgtm!!

File: src/foam/nanos/http/RefreshWebAgent.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.nanos.http;
+
+import foam.core.*;
```
Comment (kgrgreer) [inline src/foam/nanos/http/RefreshWebAgent.java]: sort imports

File: src/foam/nanos/http/RefreshWebAgent.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    out.println("<form>");
+    out.println("<label>Refresh Rate (sec):</label>");
+    out.println("<input name=\"refresh\" type=\"number\" value=\"refresh\" style=\"width:150px;display:inline-block;\"></input>");
+    out.println("<button type=submit style=\"display:inline-block;margin-top:10px;\";>Set Refresh</button>");
+    out.println("</form>");
+    out.println("</pre>");
+
+    getDelegate().execute(x);
+
+    if( json != null || !"".equals(json)){
```
Comment (kgrgreer) [inline src/foam/nanos/http/RefreshWebAgent.java]: Fix spacing.

File: src/foam/nanos/controller/ApplicationController.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
+/*
```
Comment (kgrgreer) [inline src/foam/nanos/controller/ApplicationController.js]: Why is this in this PR? Copyright header.

File: src/foam/nanos/http/RefreshWebAgent.java (hunk 1 above)
Comment (kgrgreer) [reply src/foam/nanos/http/RefreshWebAgent.java]: Still not sorted.

File: src/foam/swift/refines/Property.js [hunk 4]
```diff
@@ -617,7 +617,7 @@ if let n = newValue as? Date {
   return n
 } else if let n = newValue as? String {
   let dateFormatter = DateFormatter()
-  dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ssZZZZ"
+  dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.S'Z'"
```
Comment (kgrgreer) [inline src/foam/swift/refines/Property.js]: Why is this here?
